addons/sale_timesheet/models/account.py

```diff
@@ -70,6 +70,10 @@ def _compute_so_line(self):
     def _compute_partner_id(self):
         super(AccountAnalyticLine, self.filtered(lambda t: t._is_not_billed()))._compute_partner_id()
 
+    @api.depends('task_id', 'task_id.project_id')
```

Comment:
duplicated depends doesn't seem necessary? hr_timesheet is a requirement of sale_timesheet, so the depends should already be added by hr_timesheet.

but I guess "self.timesheet_invoice_id.state" like in the method above would make sense to add, since now the value will be dependent on `_is_not_billed`:

```py
    def _is_not_billed(self):
        return not self.timesheet_invoice_id or self.timesheet_invoice_id.state == 'cancel'
```